### PR TITLE
Refactor modifications dialogs

### PR DESCRIPTION
**Pull request by etiennehomer**

### CODE REVIEW

File: src/components/dialogs/input-hooks.js [hunk 1]
```diff
@@ -289,18 +398,200 @@ export const useConnectivityValue = ({
...
+                <Select
+                    id={label}
+                    value={value}
+                    onChange={handleChangeValue}
+                    variant="filled"
+                    fullWidth
+                    {...formProps}
+                >
+                    {enumValues.map((e) => (
+                        <MenuItem value={e.id} key={e.id}>
```
Comment (JacquesBorsenberger) [inline src/components/dialogs/input-hooks.js:556]: we are not sure that there is no duplicate id (key will be the same)

Comment (etiennehomer) [reply src/components/dialogs/input-hooks.js]: Fixed by using index

File: src/components/dialogs/input-hooks.js [hunk 2]
```diff
@@ -204,22 +292,28 @@ export const useBooleanValue = ({
...
-    clear,
+    inputForm,
     voltageLevelOptions,
     workingNodeUuid,
+    direction = 'row',
+    voltageLevelIdDefaultValue,
+    busOrBusbarSectionIdDefaultValue,
 }) => {
     const [connectivity, setConnectivity] = useState({
         voltageLevel: null,
```
Comment (laurgarn) [inline src/components/dialogs/input-hooks.js:319]: Twice `voltageLevel: null` it seems.

File: src/components/dialogs/line-creation-dialog.js [hunk 3]
```diff
@@ -4,27 +4,33 @@
...
+    filledTextField,
+    SusceptanceAdornment,
+    OhmAdornment,
+    AmpereAdornment,
+    useDoubleValue,
+    useInputForm,
+    useTextValue,
+    useConnectivityValue,
+} from './input-hooks';
+import { makeStyles } from '@material-ui/core/styles';
```
Comment (laurgarn) [inline src/components/dialogs/line-creation-dialog.js:33]: This line is just a moved from over `import { useSnackbar } from 'notistack';` . Why ?

**Review by laurgarn — APPROVED** [review]

File: src/components/dialogs/generator-creation-dialog.js [hunk 4]
```diff
@@ -277,32 +219,32 @@ const GeneratorCreationDialog = ({
...
-        setVoltageRegulation(false);
-        setVoltageSetpoint('');
-        setVoltageLevel(null);
-        setBusOrBusbarSection(null);
-    };
+    const clearValues = useCallback(() => {
+        inputForm.clear();
+    }, [inputForm]);
+
+    const handleClose = useCallback(
```
Comment (FranckLecuyer) [inline src/components/dialogs/generator-creation-dialog.js:226]: Maybe the same implementation in all dialogs and mutualize ? in another PR ...

File: src/components/dialogs/generator-creation-dialog.js [hunk 5]
```diff
@@ -277,32 +219,32 @@ const GeneratorCreationDialog = ({
...
+        (event, reason) => {
+            if (reason !== 'backdropClick') {
+                inputForm.reset();
+                onClose();
+            }
+        },
+        [inputForm, onClose]
+    );
 
     const handleCloseAndClear = () => {
```
Comment (FranckLecuyer) [inline src/components/dialogs/generator-creation-dialog.js:236]: Same as above

File: src/components/dialogs/line-creation-dialog.js (hunk 3 above)
Comment (etiennehomer) [reply src/components/dialogs/line-creation-dialog.js:34]: Fixed

File: src/components/dialogs/input-hooks.js (hunk 2 above)
Comment (laurgarn) [reply src/components/dialogs/input-hooks.js:319]: My bad. The folded common code in between fooled me.

**Review by FranckLecuyer — APPROVED** [review]

**Pull request merged**